cibuildwheel/util.py

```diff
@@ -121,7 +123,42 @@ def __repr__(self):
         return f'{self.__class__.__name__}{self.base_file_path!r})'
 
 
+class Architecture(str, Enum):
+    # mac/linux archs
+    x86_64 = 'x86_64'
+    i686 = 'i686'
+    aarch64 = 'aarch64'
+    ppc64le = 'ppc64le'
+    s390x = 's390x'
+
+    # windows archs
+    x86 = 'x86'
+    AMD64 = 'AMD64'
+
+    @staticmethod
+    def parse_config(config: str, platform: str) -> 'List[Architecture]':
+        result = []
+        for arch_str in re.split(r'[\s,]+', config):
+            if arch_str == 'auto':
+                result += Architecture.auto_archs(platform=platform)
+            else:
+                result.append(Architecture(arch_str))
```

Comment:
Do we need better error messages, either here or where it's used? Or is some stack trace + `'foo' is not a valid Architecture` good enough?

Reply:
I considered this, but the enum error message says it all, to be honest

```
(env) joerick@joerick4 ~/P/cibuildwheel> CIBW_ARCHS=x8686 cibuildwheel --platform linux          
ValueError: 'x8686' is not a valid Architecture

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/joerick/Projects/cibuildwheel/env/bin/cibuildwheel", line 11, in <module>
    load_entry_point('cibuildwheel', 'console_scripts', 'cibuildwheel')()
  File "/Users/joerick/Projects/cibuildwheel/cibuildwheel/__main__.py", line 189, in main
    archs = Architecture.parse_config(archs_config_str, platform=platform)
  File "/Users/joerick/Projects/cibuildwheel/cibuildwheel/util.py", line 145, in parse_config
    result.append(Architecture(arch_str))
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/enum.py", line 304, in __call__
    return cls.__new__(cls, value)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/enum.py", line 595, in __new__
    raise exc
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/enum.py", line 579, in __new__
    result = cls._missing_(value)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/enum.py", line 608, in _missing_
    raise ValueError("%r is not a valid %s" % (value, cls.__name__))
ValueError: 'x8686' is not a valid Architecture
```